starter/src/test/java/com/cgi/bonnie/BaseIT.java

```diff
@@ -0,0 +1,87 @@
+package com.cgi.bonnie;
+
+import com.cgi.bonnie.authentication.controller.LoginData;
+import com.cgi.bonnie.businessrules.Role;
+import com.cgi.bonnie.businessrules.user.User;
+import com.cgi.bonnie.businessrules.user.UserCredentialStorage;
+import com.cgi.bonnie.businessrules.user.UserCredentials;
+import com.cgi.bonnie.businessrules.user.UserStorage;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import io.zonky.test.db.AutoConfigureEmbeddedDatabase;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.TestInstance;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.autoconfigure.web.servlet.AutoConfigureMockMvc;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.context.annotation.Import;
+import org.springframework.kafka.test.context.EmbeddedKafka;
+import org.springframework.security.authentication.AuthenticationManager;
+import org.springframework.security.authentication.UsernamePasswordAuthenticationToken;
+import org.springframework.security.core.Authentication;
+import org.springframework.security.core.context.SecurityContext;
+import org.springframework.security.core.context.SecurityContextHolder;
+import org.springframework.test.web.servlet.MockMvc;
+import org.springframework.transaction.annotation.Transactional;
+
+import static io.zonky.test.db.AutoConfigureEmbeddedDatabase.DatabaseProvider.ZONKY;
+import static org.junit.jupiter.api.TestInstance.Lifecycle.PER_CLASS;
+import static org.springframework.boot.test.context.SpringBootTest.WebEnvironment.RANDOM_PORT;
+
+@Import(TestConfig.class)
+@Transactional
+@AutoConfigureMockMvc
+@TestInstance(PER_CLASS)
+@AutoConfigureEmbeddedDatabase(provider = ZONKY)
+@EmbeddedKafka(brokerProperties = { "listeners=PLAINTEXT://localhost:9092", "port=9092" })
+@SpringBootTest(webEnvironment = RANDOM_PORT)
+class BaseIT {
```

Comment:
In this case I'd prefer a helper class, something like `KafkaITHelper`.

Except, of course the @BeforeAll - that could go into an `@ExtendWith(KakfaITBeforeAll.class)` (Example [here](https://www.baeldung.com/junit-5-extensions#3-lifecycle-callbacks))